notebooks/finetuning.ipynb

```diff
@@ -0,0 +1,137 @@
+    "\n",
+    "Improve your Indexes by integrating your own data with Finetuning. Objective Finetuning amplifies the capabilities of AI search by incorporating your unique insights — your data and your feedback.\n",
+    "\n",
+    "Read the full [Quickstart Doc](https://www.objective.inc/docs/quickstarts/finetuning-quickstart)\n",
+    "\n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "Begin by obtaining an API key and installing the SDK:\n",
+    "\n",
+    "To get an API key, [join the waitlist](https://objective.inc/contact).\n",
+    "\n",
+    "```bash\n",
+    "pip install objective-sdk\n",
+    "```\n",
+    "\n",
+    "## Adding Data\n",
+    "\n",
+    "Begin by adding Objects to the Object Store. The code below will download 10,000 example e-commerce products and upsert them to the Object Store. In this example we are building an e-commerce search platform:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import requests\n",
+    "from objective import Client\n",
+    "\n",
+    "client = Client(api_key=\"YOUR_API_KEY\")\n",
+    "\n",
+    "data = json.loads(requests.get(\"https://d11p8vtjlacpl4.cloudfront.net/demos/ecommerce/hm-10k.json\").text)\n",
+    "\n",
+    "client.object_store.upsert_objects([{\"id\": str(obj.get(\"article_id\")), \"object\": obj} for obj in data])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Building an Index\n",
+    "\n",
+    "Build a `text` index out of the Objects pushed to the API with `searchable` and `filterable` fields:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "index = client.indexes.create_index(\n",
+    "    index_type=\"text\", # Specify the type of index as 'text'\n",
+    "    fields={\n",
+    "      # Defines fields that will be indexed, their values will influence relevance\n",
+    "      \"searchable\": [\"prod_name\", \"department_name\", \"detail_desc\", \"garment_group_name\", \"index_group_name\", \"index_name\", \"perceived_colour_master_name\", \"perceived_colour_value_name\"],\n",
+    "      # Defines fields that will be filterable\n",
+    "      \"filterable\": [\"department_name\", \"colour_group_name\", \"garment_group_name\", \"index_group_name\", \"index_name\"]\n",
+    "    }\n",
+    ")\n",
+    "index.status(watch=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Finetuned Index\n",
+    "\n",
+    "Finetuning requires feedback data. Namely, examples of queries with relevance judgements on their results. The data takes the form of `(query, object_id, label)`, where `object_id` is the ID of the search result in the Object Store, and `label` is the rating on a `{BAD, OK, GREAT}` scale. \n",
+    "\n",
+    "Here are example grades of a search for dark blue polo shirt:\n",
```

Comment:
Think it would be cool to download the data and then print the labels for "dark blue polo shirt" to more clearly connect the dots to the `training_data` data. That said, I'll leave it up to your discretion since it's a nice to have rather than a must-have.

Reply:
great idea:
<img width="637" alt="image" src="https://github.com/objective-inc/examples/assets/15079807/2a761e0e-1156-4472-bc02-dc6b16a470a7">